core/src/main/resources/db/changelog/changesets/2023_05_15_dataset_creation.yaml

```diff
@@ -0,0 +1,12 @@
+databaseChangeLog:
+  - changeSet:
+      id: "dataset_creation"
+      author: mbemis
+      changes:
+        - dropNotNullConstraint:
+            columnName:  dataset_id
```

Comment:
In that upcoming DB refactor I also want to rename this column to `tdr_dataset_id`. Datasets have 2 UUIDs (the juniper UUID for the record, and the TDR-generated UUID for the dataset). This is kind of confusing, but could be less confusing with better naming.